pkg/adaptor/cloud/azure/provider.go

```diff
@@ -232,7 +232,7 @@ func (p *azureProvider) CreateInstance(ctx context.Context, podName, sandboxID s
 
 	result, err := p.create(ctx, vmParameters)
 	if err != nil {
-		if err := p.deleteDisk(ctx, diskName); err != nil {
+		if err := p.deleteDisk(context.Background(), diskName); err != nil {
```

Comment:
unrelated?

Reply:
Yes, thought this is to minor for a separate PR, but can cherry-pick if you like.